### ISSUE TITLE
[FloatingActionButton & Background] taking default color.

### ISSUE THREAD

**Issue by Dnyaneshwar1926** [material-components/material-components-android#759]
implementation 'com.google.android.material:material:1.2.0-alpha01'

The lib taking a default color. When I add src as custom drawable(ex: image/circle drawable). It's not showing drawable instead it shows default black color & can't able to set background.

**Comment by hunterstich** [comment]
Hey there,

Thanks for taking the time to file an issue. Are you able to add some screenshots and/or a small sample that reproduces this issue so we can look into this?

**Comment by Dnyaneshwar1926** [comment]
![image](https://user-images.githubusercontent.com/49900019/68828234-03cb3c00-06cb-11ea-9d32-e82c7abb1a24.png)

**bottom bar layout.xml**

`<?xml version="1.0" encoding="utf-8"?>
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="wrap_content">

    <androidx.coordinatorlayout.widget.CoordinatorLayout
        android:layout_width="match_parent"
        android:layout_height="120dp">

        <com.google.android.material.bottomappbar.BottomAppBar
            android:id="@+id/bottom_bar_one"
            style="@style/Widget.MaterialComponents.BottomAppBar"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:layout_gravity="bottom"
            android:layout_marginStart="@dimen/margin_bottom_start_end"
            android:layout_marginEnd="@dimen/margin_bottom_start_end"
            android:layout_marginBottom="@dimen/margin_bottom_start_end"
            android:backgroundTint="@color/colorPrimary"
            app:fabCradleRoundedCornerRadius="5dp"
            tools:ignore="BottomAppBar">

            <RelativeLayout
                android:layout_width="match_parent"
                android:layout_height="match_parent">

                <androidx.appcompat.widget.AppCompatImageView
                    android:id="@+id/scheduled_bottom_bar_icon"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:layout_alignParentStart="true"
                    android:layout_marginStart="40dp"
                    app:srcCompat="@drawable/custom_fab" />

                <androidx.appcompat.widget.AppCompatImageView
                    android:id="@+id/posted_bottom_bar_icon"
                    android:layout_width="wrap_content"
                    android:layout_height="wrap_content"
                    android:layout_alignParentEnd="true"
                    android:layout_marginEnd="40dp"
                    app:srcCompat="@drawable/custom_fab" />

            </RelativeLayout>

        </com.google.android.material.bottomappbar.BottomAppBar>

        <com.google.android.material.floatingactionbutton.FloatingActionButton
            android:id="@+id/bottom_bar_fab_one"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            style="@style/Fab"
            app:srcCompat="@drawable/custom_fab"
            android:background="@drawable/bottom_bar_bg"
            app:layout_anchor="@id/bottom_bar_one"
            app:layout_anchorGravity="center" />

    </androidx.coordinatorlayout.widget.CoordinatorLayout>

</RelativeLayout>`

**Custom_fab.xml**

`<?xml version="1.0" encoding="utf-8"?>
<layer-list xmlns:android="http://schemas.android.com/apk/res/android">

    <item>
        <shape android:shape="oval">
            <stroke android:width="2dp"
                android:color="@color/white"/>
            <size android:width="50dp"
                android:height="50dp"/>
            <gradient android:startColor="@color/colorPrimaryDark"
                android:endColor="@color/fab_end_color"
                android:centerColor="@color/colorPrimary"
                android:angle="-90"/>
        </shape>
    </item>

    <item android:drawable="@drawable/ic_add"
        android:gravity="center"/>

</layer-list>`

**bottom_bar_bg.xml**

`<?xml version="1.0" encoding="utf-8"?>
<shape xmlns:android="http://schemas.android.com/apk/res/android"
    android:shape="rectangle">

    <corners
        android:bottomLeftRadius="25dp"
        android:bottomRightRadius="25dp"
        android:topLeftRadius="10dp"
        android:topRightRadius="10dp" />

    <solid android:color="@color/colorAccent" />

</shape>`


**add_icon**
![ic_add](https://user-images.githubusercontent.com/49900019/68828621-4e998380-06cc-11ea-96b4-1d9a36ffc750.png)

**Comment by hunterstich** [comment]
Thanks for the additional info. I'm not 100% sure what the expected output is, but FloatingActionButton isn't able to accept custom background drawables as it uses MaterialShapeDrawable internally to draw shadows, shape, etc. You can set app:backgroundTint on FloatingActionButton in your layout if you'd like to override the default ?attr/colorSecondary.

> Through Java Code we can set background it works but the app:fabCradleRoundedCornerRadius="5dp" between fab & BottomAppBar will not be there/support.

What are you setting the background on? A FloatingActionButton? And what is the end result? Can you attach a screenshot?

Thanks!

**Comment by Dnyaneshwar1926** [comment]
Here is a screenshot. I set custom drawable which is showing black color at center. 
Yeah, we can set background tint but I wanna make gradient colored fab. So for this, I need to set background.  

**Setting background code**
bottomBarFabOne.setBackground(getResources().getDrawable(R.drawable.custom_fab));

As I'm setting my background through jave/xml code nothing gets an effect on the background. We can only set background tint. As I observed in materialDefaultTheme background tag is @null. 

**result**
![image](https://user-images.githubusercontent.com/49900019/68918183-33457b80-0793-11ea-9fbe-70b0fdb0fdcb.png)


**question:**
How to set the corner radius for BottomAppBar?
ex:- leftTopRadius, rightBottomRadius etc

**expected result**
![image](https://user-images.githubusercontent.com/49900019/68918981-532a6e80-0796-11ea-9812-30d65c9e9602.png)
